consensus/src/dag/reliable_broadcast.rs

```diff
@@ -184,3 +186,53 @@ impl RpcHandler for NodeBroadcastHandler {
         }
     }
 }
+
+#[derive(Debug, ThisError)]
+pub enum CertifiedNodeHandleError {
+    #[error("node already exists")]
+    NodeExists,
+    #[error("missing parents")]
+    MissingParents,
+}
+
+pub struct CertifiedNodeHandler {
+    dag: Arc<RwLock<Dag>>,
+}
+
+impl CertifiedNodeHandler {
+    pub fn new(dag: Arc<RwLock<Dag>>) -> Self {
+        Self { dag }
+    }
+}
+
+impl RpcHandler for CertifiedNodeHandler {
+    type Request = CertifiedNode;
+    type Response = CertifiedAck;
+
+    fn process(&mut self, node: Self::Request) -> anyhow::Result<Self::Response> {
+        {
+            let dag_reader = self.dag.read();
+            if dag_reader.exists(&node.digest()) {
+                bail!(CertifiedNodeHandleError::NodeExists);
+            }
+
+            if node.metadata().round() > 0
+                && (node.parents().len() == 0
```

Comment:
I am not sure you need this check.
If the certificate is valid then f+1 honest signed it. 
If an honest signed it then node has enough parents.